### PR TITLE
fixing compilation fallback

### PR DESCRIPTION
**Pull request by jjsjann123**
We need to canonicalize graph representation before hash it to unregister. Otherwise, we are not removing the right kernel_id in graph_cache

### CODE REVIEW

**Review by kevinstephano — APPROVED** [review]
LGTM

**Comment by jjsjann123** [comment]
Please do not merge yet. I'll run tests later and merge then.

**Comment by jjsjann123** [comment]
tests passed. waiting on CI to push the merge button.

**Pull request merged**
